### ISSUE TITLE
Markdown support in the future?

### ISSUE THREAD

**Issue by JPFlare** [lmg-anon/mikupad#94]
It seems like it's probably a big undertaking, but I do think Mikupad would benefit from having markdown support. There are lots of situations where you might want text to be bold, italicized, underscored, etc. Support for having bold and italic text would also work, but as far as I know, the backend has no way of making the text is bold or italicized except with markdown formatting.

I did a quick issue search, and I couldn't find a direct mention of requesting this be added as a feature, so I figured I would open one. It's not a huge deal presently, as you can just have the LLM write as if it were using markdown and export the text to read in another editor, but it makes sense to me to request it as a feature if it might be widely used.

The ideal implementation may be a toggle under Editor Preferences, depending on if the average user might find it cumbersome or not, but I'm not sure how feasible that is.

**Comment by lmg-anon** [comment]
Yes, I agree. I'm not sure how markdown support could be implemented while keeping it editable, but maybe it could be done differently. For example, there could be two editor views, one for editing and one for markdown preview, with a button to switch between them. Alternatively, the editor could be split, with the normal Mikupad view on the left and the markdown preview on the right.

**Comment by JPFlare** [comment]
If you take a look at how this site does it, it is possible to have it all in a single editor, it just makes the markdown syntax less visible: https://stackedit.io/app#

It depends if it's an acceptable tradeoff for most users, I think. Having a separate panel with a preview would also work though, it would just be harder to work with for editing.

**Comment by neCo2** [comment]
I believe this "mixed" implementation in stackedit is very readable.
I don't remember where, but I've also seen a markdown editor once, where it applies the formatting, and fully hides the md syntax. I don't think this makes for a sensible default, but it could be considered as an option.

**Comment by jukofyork** [comment]
This type of interface with a simple method to switch between editor and preview is very easy to use:

![editor-overview](https://github.com/user-attachments/assets/38b15734-301e-4b6d-ad93-8e475afd122e)

https://model-realtime.hcldoc.com/help/index.jsp?topic=%2Forg.eclipse.mylyn.wikitext.help.ui%2Fhelp%2FGetting-Started.html
